### DISCUSSION TITLE
is it possible to use server actions with react-query library ?

### DISCUSSION THREAD

**Discussion by jaimeenkoffeekodes** [vercel/next.js#62042] [Help]
### Summary

in below code, right now its fully client component, so question is that is it possible to make fetchData as server action ? if yes then will it effect importing useQuery because when we use hooks, it should be client component, if no then is there any best way to leverage next.js feature in my api calling?

### Additional information

```js
import React from 'react';
import useFetchData from './useFetchData';

const fetchData = async () => {
  const response = await fetch('https://############');
  if (!response.ok) {
    throw new Error('Network response was not ok');
  }
  return response.json();
};

const YourComponent = () => {
  const { isLoading, isError, data: posts, error } = useFetchData(['posts'], fetchData);

  if (isLoading) return <div>Loading...</div>;
  if (isError) return <div>Error: {error.message}</div>;

  return (
    <div>
      {/* Render your posts data here */}
    </div>
  );
};

export default YourComponent;
```

**Comment by mutasim77** [comment]
The Tanstack Query docs offers a complete [guide](https://tanstack.com/query/v5/docs/framework/react/guides/advanced-ssr) on this subject. Additionally, check this [video](https://www.youtube.com/watch?v=yVsaCVEfPn4) also

**Reply by jaimeenkoffeekodes** [reply]
thanks!
